File: 3447-clear-digits/3447-clear-digits.go
func clearDigits(s string) string {
    stack := []byte{}

	for i := range s {

		if s[i] >= 65 && s[i] <= 90 || s[i] >= 97 && s[i] <= 122 {
			stack = append(stack, s[i])
		} else {
			if len(stack) > 0 {
				stack = stack[:len(stack)-1]
			}

		}

	}

	return string(stack)
}

File: 3447-clear-digits/3447-clear-digits.py
class Solution(object):
    def clearDigits(self, s):
        """
        :type s: str
        :rtype: str
        """
        stack = []
        for char in s:
            if not char.isdigit():
                stack.append(char)
            elif stack:
                stack.pop()
        return ''.join(stack)